pex/resolve/lock_resolver.py

```diff
@@ -117,7 +117,9 @@ def resolve_from_lock(
         )
         parsed_requirements = try_(
```

Comment:
This is all fine and works, but you've also pushed a concern of `lock_resolver.py` to lots of other places. It's not clear to me anyway that users of all those places have the same concerns. The Error you return from deeper down definitely makes sense when its returned from right here, I'm not sure about other sites. Did you consider just ~:
```python
if not parsed_requirements.requirements:
    parsed_requirements = Requirements.create(
        parsed_requirements=parse_requirement_strings(
            str(req) for req in lock.requirements
        ),
        parsed_constraints=parsed_requirements.parsed_constraints
    )

```

Reply:
You're right: `pex3 lock create` is using `parse_lockable_requirements` and this PR resulted in that now erroring when no requirements are specified. I do think we should error (https://github.com/pantsbuild/pex/issues/1659), but this is a bad error message. Callers should raise the error.

Re doing the parsing in `lock_resolver.py` directly, I started with that, but realized we want the logic to filter out VCS and local requirements. I took your suggestion though and reverted changes to `requirement_configuration.py`.

Thanks!

Reply:
Since filtering already occurs before a lock is created, the requirements written down there to be used as fallback are kosher. MyPy is telling you a lie this time!

Reply:
I guess my point is the coversion from ParsedRequirement is lossy on the typing side but it never actually matters since we vet the ParsedRequirement types before ever writing them as Requirements in the lockfile; so MyPy will always be noise here.